lib/manageiq/providers/openstack/legacy/events/openstack_stf_event_converter.rb

```diff
@@ -0,0 +1,30 @@
+class OpenstackStfEventConverter
+  def initialize(event)
+    @event = event.symbolize_keys
+    @event_payload = @event.fetch(:payload).symbolize_keys
+    @payload = hashize_traits(@event_payload.fetch(:traits))
+  end
+
+  def metadata
+    {:user_id => nil, :priority => nil, :content_type => nil}
+  end
+
+  def payload
+    {
+      "message_id" => @event_payload.fetch(:message_id),
+      "event_type" => @event_payload.fetch(:event_type),
+      "timestamp"  => @event_payload.fetch(:generated),
+      "payload"    => @payload,
+    }
+  end
+
+  private
+
+  def hashize_traits(traits_list)
+    output = {}
+    traits_list.each do |trait_elem|
+      output[trait_elem.first] = trait_elem.last
+    end
```

Comment:
@aufi could this just be `Hash[@event_payload.fetch(:traits)]` ?

Reply:
The ```:traits``` value is array of 3 element arrays, so it doesn't work. (TBH I don't know why is there the middle element "1", but it is there..)
```
"traits"=>[
            ["service", 1, "network.controller-0.redhat.local"],
            ["request_id", 1, "req-2520570f-7549-414c-98b2-d64ad7e0ffa0"],
            ["project_id", 1, "9609072e9f3b401dac5f7ace13672627"],
            ["user_id", 1, "e4c58a343ce845bbb61fc0b435c5b434"],
            ["tenant_id", 1, "9609072e9f3b401dac5f7ace13672627"],
            ["name", 1, "stfnet7"]
           ],
```

Reply:
Ah okay :+1: